plugins/modules/packaging/os/zypper.py

```diff
@@ -336,7 +337,7 @@ def get_cmd(m, subcommand):
     "puts together the basic zypper command arguments with those passed to the module"
     is_install = subcommand in ['install', 'update', 'patch', 'dist-upgrade']
     is_refresh = subcommand == 'refresh'
-    cmd = ['/usr/bin/zypper', '--quiet', '--non-interactive', '--xmlout']
+    cmd = [find_executable('zypper'), '--quiet', '--non-interactive', '--xmlout']
```

Comment:
Please use `m.get_bin_path` instead (https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1406). Same for `zypper_repository` (you'll need to pass the `AnsibleModule` instance to `_get_cmd()` there though).

You also need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

Reply:
Thank you for your review! I improved the PR according to your recommendations.